apiserver/v1/types_apirequestcount.go

```diff
@@ -41,59 +47,95 @@ type APIRequestCountStatus struct {
 	// +patchStrategy=merge
 	Conditions []metav1.Condition `json:"conditions"`
 
-	// requestsLastHour contains request history for the current hour. This is porcelain to make the API
+	// removedInRelease is when the API will be removed.
+	// +kubebuilder:validation:Pattern=^[0-9][0-9]*\.[0-9][0-9]*$
+	// +kubebuilder:validation:MinLength=3
+	// +kubebuilder:validation:MaxLength=64
+	// +optional
+	RemovedInRelease string `json:"removedInRelease"`
+
+	// requestCount is a sum of all requestCounts across all current hours, nodes, and users.
+	// +kubebuilder:validation:Minimum=0
+	// +required
+	RequestCount int64 `json:"requestCount"`
+
+	// requestsInTheCurrentHour contains request history for the current hour. This is porcelain to make the API
 	// easier to read by humans seeing if they addressed a problem. This field is reset on the hour.
-	RequestsLastHour PerResourceAPIRequestLog `json:"requestsLastHour"`
+	// +optional
+	RequestsLastHour PerResourceAPIRequestLog `json:"requestsInTheCurrentHour"`
```

Comment:
what happens on master replacement? Anybody cleaning this up? Same for 24h below.

Reply:
> what happens on master replacement? Anybody cleaning this up? Same for 24h below.

Yes.  The controller does so.  It always clears the oldest 24h block.

Reply:
which controller? In kube-apiserver? That does not know about the set of nodes.